### DISCUSSION TITLE
What happend to grounding in description layer and what is quality and is it used??

### DISCUSSION THREAD

**Discussion by flomickl** [apache/streampipes#1173] [Dev]
Hi
still working on the processor documentation.
During this process, I was wondering that the functions of the groundings are no longer used in the declairMethod.

So this "old" picture is out of date, I guess.
Was it replaced with something else or now totally in the backend?

![Datalayer in streampipes](https://user-images.githubusercontent.com/16739072/215327380-56fe498f-5fe4-4752-a4c3-c00c2ff94d6d.png)

What happend to it, why and why are the method then still integrated?

.supportedFormats(TransportFormat... format)
supportedFormats(List< TransportFormat > formats)

supportedProtocols(TransportProtocol... protocol)
supportedProtocols(List< TransportProtocol> protocols)

Furthermore
I did not find any examples of the quantity layer.
The only part I found was in the meassermentURI, but I guess this is also part of the Event Schema itself?

I want to create a newer version of it:
![data_and:description_layer](https://user-images.githubusercontent.com/16739072/215327668-262de022-8931-4f9f-aff8-9fe231304e07.png)

How would you describe it now?

**Comment by dominikriemer** [comment]
Grounding is still being used, but we typically don't define it directly in the data processor anymore, but in the `provideServiceDefinition` method of the submitter to ease the defintion of pipeline elements. So when registering a messaing fromat or protocol, the description is automatically added to the processor.
Quality has been removed for now, as we didn't make any real use of it and were trying to simplify the codebase.

**Comment by flomickl** [comment]
ah good to know.
So for now, the event schema is the only layer.

and these are the variables from the event schema that are still used?

| SP Class            | Name                           | Type      | Description |
| ------------------- | ------------------------------ | --------- | ---------- |
| Class EventProperty | label                          | String    |            |
| Class EventProperty | description                    | String    |            |
|                     | measurementUnit                | String    |            |
| Class EventProperty | boolean required               | boolean   |            |
| Class EventProperty | propertyScope                  | String    |            |
| Class EventProperty | index                          | int       |            |
| Class EventProperty | runtimeID                      | String    |            |


and this variables are not in use?
| spcLASS             | Name                           | Type      |     |
| ------------------- | ------------------------------ | --------- | --- |
| Class EventProperty | eventPropertyQualities         | ArrayList |     |
| Class EventProperty | requiresEventPropertyQualities | ArrayList |     |
|                     | String valuesSpecitication     | String    |     |

**Comment by flomickl** [comment]
We should update then this visualizations

![grafik](https://user-images.githubusercontent.com/16739072/216394743-c6e80248-76ed-40f7-8e1e-7645e6fca642.png)

here
https://streampipes.apache.org/docs/docs/technicals-architecture.html


also mentioned here: https://github.com/apache/streampipes/discussions/1184#discussioncomment-4854033
